### PR TITLE
Prefer jstor to jstor doi which is often broken

### PR DESCRIPTION
**Pull request by GlazerMann**

### CODE REVIEW

File: Template.php [hunk 1]
```diff
@@ -858,6 +858,10 @@ protected function get_identifiers_from_url($url_sent = NULL) {
         $this->set('url', $url); // Save it
       }
     }
+   
+    if (preg_match('~^(https?://[www.|]jstor.org/stable/)(10.2307/)(.*)$~', $url, $matches)) {
```
Comment (ms609) [inline Template.php]: - [www.|] will match w, w, w, or anything.  Something close to  `(?:www\.)` is probably what you intend
- escape the `.` in jstor.org
- Why do we need to capture (10.2307/)?  Remove parentheses?
- `.+` should probably replace `.*` in final capture set; we want a non-empty JSTOR id.

Comment (ms609) [reply Template.php]: Is there any value in adding a test, too?  Possibly a test will duplicate existing ones but at the least a test would have caught some of these.

Comment (GlazerMann) [reply Template.php]: RegEx fixed.  Will add test
